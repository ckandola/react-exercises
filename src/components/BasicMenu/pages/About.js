import React from 'react';
import '../BasicMenu.css';

const About = () => {
    return (
        <div className="basicmenu-about">
            <h3>About Me</h3>
            <p>
                I studied programming in Python and Java when I was at Whatcom Community College, and I transferred to Western 
                Washington University to pursue a Bachelor's degree in computer science. My school projects were usually just 
                tests about my understanding of programming basics, but I did program a few games to demonstrate my knowledge. 
                I've made a software version of a board game called Deadwood, a Carmen Sandiego app for Android, a driving game, 
                and a stealth game using object-oriented languages like Java and C#.
            </p>

            <p>
                I had a few classes in MySQL, C, and Racket as well, but they were primarily for learning about databases, algorithms,
                and operating systems like Linux. I had several projects like simulating database behavior in Java classes, simulating
                shell behavior in C, and creating servers and clients.
            </p>
            
            <p>
                Since graduating, I have had two internships in software development, which mostly dealt with Javascript. I learned about
                Node.js, Chrome's Dev Tools, Git, and about working remotely in general in my first internship, which was with Cornerstone
                Systems Northwest. After learning a decent amount of Javascript, I was tasked with creating an encryption/decryption 
                module for my team's secret environment variables. 
            </p>
            
        </div>
    );
}

export default About;
