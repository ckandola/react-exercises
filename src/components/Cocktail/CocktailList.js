import React from 'react';
import Cocktail, { useGlobalContext } from './Cocktail';
import Loading from './Loading';

const CocktailList = () => {

    const { cocktails, cocktailLoading } = useGlobalContext();

    if (cocktailLoading) {
        return <Loading />;
    }

    if (cocktails.length < 1) {
        return (
            <h2 className="cocktail-section-title">
                No cocktails match your search
            </h2>
        );
    }

    return (
        <section className="cocktails-section">
            <h2 className="cocktail-section-title">Cocktails</h2>
            <div className="cocktails-center">
                {cocktails.map(drink => {
                    return (
                        <Cocktail key={drink.id} {...drink} />
                    );
                })}
            </div>
        </section>
    );
};

export default CocktailList;
