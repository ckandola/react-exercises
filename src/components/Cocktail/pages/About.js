import React from 'react';

const About = () => {
    return (
        <section className="cocktails-section cocktail-about-section">
            <h1 className="cocktail-section-title">
                About Us
            </h1>
            <p>Un jour je serai le meilleur dresseur. Je me
                battrai sans répit. Je ferai tout pour être
                vainqueur et gagner les défis. Je parcourrai
                la terre entière tracquant avec espoir les
                pokémons et leurs mystères, le secret de leurs
                pouvoirs. C'est notre histoire! Ensemble pour
                la victoire! Rien ne nous arrêtera; notre 
                amitié triomphera. Même à notre âge, un voyage
                d'apprentissage. Ça demande du courage.
                Rien ni personne sur mon chemin ne pourra me 
                briser car pour accomplir mon destin je suis
                déterminé. Quand il faudra passer à l'action
                nos amis seront là. Et tous ensemble nous
                gagnerons la dernière des combats.
            </p>
        </section>
    );
};

export default About;
