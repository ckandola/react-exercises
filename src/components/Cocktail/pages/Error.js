import React from 'react';
import { Link } from 'react-router-dom';

const Error = () => {
    return (
        <section className="cocktails-section cocktail-error-page">
            <div className="cocktail-error-container">
                <h3>Oops! That page doesn't exist</h3>
                <Link to="/Cocktail" className="cocktail-btn-primary">
                    Back Home
                </Link>
            </div>
        </section>
    );
};

export default Error;
