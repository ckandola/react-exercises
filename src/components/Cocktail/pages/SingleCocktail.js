import React from 'react';
import Loading from '../Loading';
import { useParams, Link } from 'react-router-dom';
const url = 'https://www.thecocktaildb.com/api/json/v1/1/lookup.php?i=';

const SingleCocktail = () => {
    const {id} = useParams();
    const [loading, setLoading] = React.useState(false);
    const [cocktail, setCocktail] = React.useState(null);

    React.useEffect(() => {
        setLoading(true);
        const getCocktail = async () => {
            try {
                const response = await fetch(`${url}${id}`);
                const data = await response.json();
                if (data.drinks) {
                    const {strDrink: name, 
                        strDrinkThumb: img, 
                        strAlcoholic: isAlcoholic, 
                        strCategory: category, 
                        strGlass: glass, 
                        strInstructions: instructions, 
                        strIngredient1,
                        strIngredient2,
                        strIngredient3,
                        strIngredient4,
                        strIngredient5,
                    } = data.drinks[0];
                    const ingredients = [
                        strIngredient1,
                        strIngredient2,
                        strIngredient3,
                        strIngredient4,
                        strIngredient5
                    ];
                    const newCocktail = {name, img, isAlcoholic, category, glass, instructions, ingredients};
                    setCocktail(newCocktail);
                } else {
                    setCocktail(null);
                }
                setLoading(false);
            } catch (error) {
                console.error(error);
                setLoading(false);

            }
        }
        getCocktail();
    }, [id]);

    if (loading) {
        return (
            <Loading />
        );
    };

    if (!cocktail) {
        return (
            <div className="cocktail-error-container">
                <h2 className="cocktail-section-title">No cocktail to display</h2>
                <Link to="/Cocktail" className="cocktail-btn-primary">Back Home</Link>
            </div>
        );
    };

    const {name, img, category, isAlcoholic, glass, ingredients, instructions} = cocktail;
    return (
        <section className="cocktails-section cocktail-section">
            <Link to={"/cocktail"} className="cocktail-btn cocktail-btn-primary">Back Home</Link>
            <h2 className="cocktail-section-title">{name}</h2>
            <div className="cocktail-drink">
                <img src={img} alt={name}/>
                <div className="cocktail-drink-info">
                    <p>
                        <span className="cocktail-drink-data">
                            Name: {name}
                        </span>
                    </p>
                    <p>
                        <span className="cocktail-drink-data">
                            category: {category}
                        </span>
                    </p>
                    <p>
                        <span className="cocktail-drink-data">
                            info: {isAlcoholic}
                        </span>
                    </p>
                    <p>
                        <span className="cocktail-drink-data">
                            glass: {glass}
                        </span>
                    </p>
                    <p>
                       <span className="cocktail-drink-data">
                            instructions: {instructions}
                        </span>
                    </p>
                    <p>
                        <span className="cocktail-drink-data">
                            Ingredients: {ingredients.map((ingredient, index) => {
                                return (ingredient && <span key={index}>{ingredient}</span>);
                            })}
                        </span>
                    </p>
                </div>
            </div>
        </section>
    );
};

export default SingleCocktail;
