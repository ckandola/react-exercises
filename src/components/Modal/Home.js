import React from 'react';
import {useGlobalContext} from '../../context';
import './Modal.css';

const Home = () => {
    const {openModal} = useGlobalContext();

    return (
        <main className="modal-main">
            <button 
                className="modal-btn"
                onClick={openModal}>
                Show Modal
            </button>
        </main>
    );
};

export default Home;
