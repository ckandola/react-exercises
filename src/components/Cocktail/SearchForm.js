import React from 'react';
import { useGlobalContext } from './Cocktail';

const SearchForm = () => {
    return (
        <div>
            <h2>SearchForm</h2>
        </div>
    );
};

export default SearchForm;
