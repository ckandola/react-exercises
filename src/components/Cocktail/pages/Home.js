import React from 'react';
import CocktailList from '../CocktailList';
import SearchForm from '../SearchForm';

const Home = () => {
    return (
        <div>
            <h2>Home Page</h2>
        </div>
    );
};

export default Home;
