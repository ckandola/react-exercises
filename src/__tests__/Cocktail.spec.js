import React, { useRef } from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as AppProvider from '../context';
import {BrowserRouter, MemoryRouter} from 'react-router-dom';

import CocktailMain from '../components/Cocktail';
import Loading from '../components/Cocktail/Loading';
import NavBar from '../components/Cocktail/NavBar';
import SearchForm from '../components/Cocktail/SearchForm';
import CocktailList from '../components/Cocktail/CocktailList';

import Home from '../components/Cocktail/pages/Home';
import About from '../components/Cocktail/pages/About';
import Error from '../components/Cocktail/pages/Error';
import SingleCocktail from '../components/Cocktail/pages/SingleCocktail';

import {waitFor} from '@testing-library/react';

jest.mock('react', () => {
    const originReact = jest.requireActual('react');
    const mUseRef = jest.fn();
    return {
        ...originReact,
        useRef: mUseRef
    }
});

describe('Cocktail tests', () => {
    
    const placeholderFn = () => {};
    describe('snapshots match', () => {

        test('CocktailMain matches the snapshot', () => {
            const contextValues = {
                cocktailLoading: false, cocktails: [], setCocktailSearchTerm: placeholderFn
            };
            jest.spyOn(AppProvider, 'useGlobalContext').mockImplementation(() => contextValues);
            const wrapper = mount(<CocktailMain />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        test('Loading matches the snapshot', () => {
            const wrapper = shallow(<Loading />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        test('NavBar matches the snapshot', () => {
            const wrapper = shallow(<NavBar />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        
        test('SearchForm matches the snapshot', () => {
            const contextValues = {
                cocktailLoading: false, cocktails: [], setCocktailSearchTerm: placeholderFn
            };
            const mRef = {current: {}};
            useRef.mockReturnValueOnce(mRef);
            jest.spyOn(AppProvider, 'useGlobalContext').mockImplementation(() => contextValues);
            const wrapper = mount(<SearchForm />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        test('CocktailList and Cocktail match the snapshot', () => {
            const contextValues = {
                cocktailLoading: false, 
                cocktails: [{img: 'img', name: 'name', id: '0', isAlcoholic: 'Non-alcoholic', glass: 'glass'}],
                setCocktailSearchTerm: placeholderFn
            };
            jest.spyOn(AppProvider, 'useGlobalContext').mockImplementation(() => contextValues);
            const wrapper = mount(
                <BrowserRouter>
                    <CocktailList />
                </BrowserRouter>
            );
            expect(toJson(wrapper)).toMatchSnapshot();
            expect(wrapper.find('Cocktail')).toHaveLength(1);
        });

        test('Home and SingleCocktail components match the snapshot', () => {
            const contextValues = {
                cocktailLoading: false,
                cocktails: [{img: 'img', name: 'name', id: '0', isAlcoholic: 'Non-alcoholic', glass: 'glass'}],
                setCocktailSearchTerm: placeholderFn
            };
            jest.spyOn(AppProvider, 'useGlobalContext').mockImplementation(() => contextValues);
            const mRef = {current: {}};
            useRef.mockReturnValueOnce(mRef);
            const wrapper = mount(
                <BrowserRouter>
                    <Home />
                </BrowserRouter>
            );
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        test('About page matches the snapshot', () => {
            const wrapper = shallow(<About />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        
        test('Error page matches the snapshot', () => {
            const wrapper = shallow(<Error />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    test('SearchForm calls `setCocktailSearchTerm` when search input is changed', () => {
        const mRef = {current: {}};
        useRef.mockReturnValueOnce(mRef);
        const mockSetTerm = jest.fn();
        const contextValues = {
            cocktailLoading: false,
            cocktails: [{img: 'img', name: 'name', id: '0', isAlcoholic: 'Non-alcoholic', glass: 'glass'}],
            setCocktailSearchTerm: mockSetTerm
        };
        jest.spyOn(AppProvider, 'useGlobalContext').mockImplementation(() => contextValues);
        const wrapper = mount(<SearchForm />);
        expect(mockSetTerm.mock.calls.length).toBe(0);
        wrapper.find('input').simulate('change', {target: {value: 'test'}});
        expect(mockSetTerm.mock.calls.length).toBe(1);
    });

    test('if `cartLoading` is true, then CocktailList renders Loading component', () => {
        const contextValues = {
            cocktailLoading: true, 
            cocktails: [{img: 'img', name: 'name', id: '0', isAlcoholic: 'Non-alcoholic', glass: 'glass'}],
            setCocktailSearchTerm: placeholderFn
        };
        jest.spyOn(AppProvider, 'useGlobalContext').mockImplementation(() => contextValues);
        const wrapper = mount(
            <BrowserRouter>
                <CocktailList />
            </BrowserRouter>
        );
        expect(wrapper.find('Loading')).toHaveLength(1);
    });

    test('if `cocktails` is empty array, CocktailList returns "No cocktails match your search"', () => {
        const contextValues = {
            cocktailLoading: false, 
            cocktails: [],
            setCocktailSearchTerm: placeholderFn
        };
        jest.spyOn(AppProvider, 'useGlobalContext').mockImplementation(() => contextValues);
        const wrapper = mount(
            <BrowserRouter>
                <CocktailList />
            </BrowserRouter>
        );
        expect(wrapper.find('.cocktail-section-title').getElement().props.children).toEqual('No cocktails match your search');
    });

    test('navigation to a non-existent page will lead to the Error page', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/cocktail_this-page-does-not-exist']}>
                <CocktailMain />
            </MemoryRouter>
        );
        expect(wrapper.find('SingleCocktail')).toHaveLength(0);
        expect(wrapper.find('Error')).toHaveLength(1);
    });

    /* THIS DOES NOT WORK - EITHER IS STUCK LOADING, OR WILL NOT 'FETCH' ANYTHING
    test('navigation to the id of an existing cocktail will lead to the appropriate SingleCocktail page', async () => {
        // https://jaketrent.com/post/mock-fetch-jest-test

        const fakeData = {
            drinks: [
                {idDrink: '11690', strDrink: 'Mai Tai', strDrinkThumb: 'https://www.thecocktaildb.com/images/media/drink/twyrrp1439907470.jpg', 
                strAlcoholic: 'Alcoholic', strCategory: 'ordinary drink', 
                strGlass: 'Collins glass', 
                strInstructions: "I don't care - it's just a test anyway", 
                strIngredient1: 'light rum'}
            ]
        };

        function setUpFetchStub () {
            console.log('I WAS SUMMONED')
            return new Promise((resolve, reject) => {
                process.nextTick(() =>
                    true ? resolve({json: () => Promise.resolve({fakeData})})
                    : reject({error: 'oops'})
                )
            })
        };
        
        jest.spyOn(console, 'error').mockImplementation(() => {});
        jest.spyOn(global, 'fetch').mockImplementation(setUpFetchStub());
        const wrapper = mount(
            <MemoryRouter initialEntries={['/cocktail/11690']}>
                <SingleCocktail />
            </MemoryRouter>
        );
        expect(wrapper.find('Loading')).toHaveLength(1);
        // console.log(wrapper.debug())
        await waitFor(() => expect(wrapper.find('.cocktail-drink-data')).toHaveLength(6));
        global.fetch.mockClear();
    }); */
});
