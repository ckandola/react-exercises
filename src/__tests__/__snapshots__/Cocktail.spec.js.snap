// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cocktail tests snapshots match About page matches the snapshot 1`] = `
<section
  className="cocktails-section cocktail-about-section"
>
  <h1
    className="cocktail-section-title"
  >
    About Us
  </h1>
  <p>
    Un jour je serai le meilleur dresseur. Je me battrai sans répit. Je ferai tout pour être vainqueur et gagner les défis. Je parcourrai la terre entière tracquant avec espoir les pokémons et leurs mystères, le secret de leurs pouvoirs. C'est notre histoire! Ensemble pour la victoire! Rien ne nous arrêtera; notre amitié triomphera. Même à notre âge, un voyage d'apprentissage. Ça demande du courage. Rien ni personne sur mon chemin ne pourra me briser car pour accomplir mon destin je suis déterminé. Quand il faudra passer à l'action nos amis seront là. Et tous ensemble nous gagnerons la dernière des combats.
  </p>
</section>
`;

exports[`Cocktail tests snapshots match CocktailList and Cocktail match the snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CocktailList>
      <section
        className="cocktails-section"
      >
        <h2
          className="cocktail-section-title"
        >
          Cocktails
        </h2>
        <div
          className="cocktails-center"
        >
          <Cocktail
            glass="glass"
            id="0"
            img="img"
            isAlcoholic="Non-alcoholic"
            key="0"
            name="name"
          >
            <article
              className="cocktail"
            >
              <img
                alt="name"
                className="cocktail-img"
                src="img"
              />
              <div
                className="cocktail-footer"
              >
                <h3>
                  name
                </h3>
                <h4>
                  glass
                </h4>
                <p>
                  Non-alcoholic
                </p>
                <Link
                  className="cocktail-btn cocktail-btn-primary cocktail-btn-details"
                  to="/cocktail/0"
                >
                  <LinkAnchor
                    className="cocktail-btn cocktail-btn-primary cocktail-btn-details"
                    href="/cocktail/0"
                    navigate={[Function]}
                  >
                    <a
                      className="cocktail-btn cocktail-btn-primary cocktail-btn-details"
                      href="/cocktail/0"
                      onClick={[Function]}
                    >
                      Details
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </article>
          </Cocktail>
        </div>
      </section>
    </CocktailList>
  </Router>
</BrowserRouter>
`;

exports[`Cocktail tests snapshots match CocktailMain matches the snapshot 1`] = `
<CocktailMain>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Navbar>
        <nav
          className="cocktail-navbar"
        >
          <div
            className="cocktail-nav-center"
          >
            <Link
              to="/Cocktail"
            >
              <LinkAnchor
                href="/Cocktail"
                navigate={[Function]}
              >
                <a
                  href="/Cocktail"
                  onClick={[Function]}
                >
                  <img
                    alt="cocktail logo"
                    className="cocktail-logo"
                    src="logo.svg"
                  />
                </a>
              </LinkAnchor>
            </Link>
            <ul
              className="cocktail-nav-links"
            >
              <li>
                <Link
                  to="/Cocktail"
                >
                  <LinkAnchor
                    href="/Cocktail"
                    navigate={[Function]}
                  >
                    <a
                      href="/Cocktail"
                      onClick={[Function]}
                    >
                      Home
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
              <li>
                <Link
                  to="/cocktail_about"
                >
                  <LinkAnchor
                    href="/cocktail_about"
                    navigate={[Function]}
                  >
                    <a
                      href="/cocktail_about"
                      onClick={[Function]}
                    >
                      About
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
            </ul>
          </div>
        </nav>
      </Navbar>
      <Switch>
        <Route
          computedMatch={
            Object {
              "isExact": true,
              "params": Object {
                "0": "/",
              },
              "path": "*",
              "url": "/",
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          path="*"
        >
          <Error>
            <section
              className="cocktails-section cocktail-error-page"
            >
              <div
                className="cocktail-error-container"
              >
                <h3>
                  Oops! That page doesn't exist
                </h3>
                <Link
                  className="cocktail-btn-primary"
                  to="/Cocktail"
                >
                  <LinkAnchor
                    className="cocktail-btn-primary"
                    href="/Cocktail"
                    navigate={[Function]}
                  >
                    <a
                      className="cocktail-btn-primary"
                      href="/Cocktail"
                      onClick={[Function]}
                    >
                      Back Home
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </section>
          </Error>
        </Route>
      </Switch>
    </Router>
  </BrowserRouter>
</CocktailMain>
`;

exports[`Cocktail tests snapshots match Error page matches the snapshot 1`] = `
<section
  className="cocktails-section cocktail-error-page"
>
  <div
    className="cocktail-error-container"
  >
    <h3>
      Oops! That page doesn't exist
    </h3>
    <Link
      className="cocktail-btn-primary"
      to="/Cocktail"
    >
      Back Home
    </Link>
  </div>
</section>
`;

exports[`Cocktail tests snapshots match Home and SingleCocktail components match the snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Home>
      <main
        className="none"
      >
        <SearchForm>
          <section
            className="cocktails-section cocktail-search"
          >
            <form
              className="cocktail-search-form"
              onSubmit={[Function]}
            >
              <div
                className="cocktail-form-control"
              >
                <label
                  htmlFor="name"
                >
                  Search your favorite cocktail
                </label>
                <input
                  id="name"
                  onChange={[Function]}
                  type="text"
                />
              </div>
            </form>
          </section>
        </SearchForm>
        <CocktailList>
          <section
            className="cocktails-section"
          >
            <h2
              className="cocktail-section-title"
            >
              Cocktails
            </h2>
            <div
              className="cocktails-center"
            >
              <Cocktail
                glass="glass"
                id="0"
                img="img"
                isAlcoholic="Non-alcoholic"
                key="0"
                name="name"
              >
                <article
                  className="cocktail"
                >
                  <img
                    alt="name"
                    className="cocktail-img"
                    src="img"
                  />
                  <div
                    className="cocktail-footer"
                  >
                    <h3>
                      name
                    </h3>
                    <h4>
                      glass
                    </h4>
                    <p>
                      Non-alcoholic
                    </p>
                    <Link
                      className="cocktail-btn cocktail-btn-primary cocktail-btn-details"
                      to="/cocktail/0"
                    >
                      <LinkAnchor
                        className="cocktail-btn cocktail-btn-primary cocktail-btn-details"
                        href="/cocktail/0"
                        navigate={[Function]}
                      >
                        <a
                          className="cocktail-btn cocktail-btn-primary cocktail-btn-details"
                          href="/cocktail/0"
                          onClick={[Function]}
                        >
                          Details
                        </a>
                      </LinkAnchor>
                    </Link>
                  </div>
                </article>
              </Cocktail>
            </div>
          </section>
        </CocktailList>
      </main>
    </Home>
  </Router>
</BrowserRouter>
`;

exports[`Cocktail tests snapshots match Loading matches the snapshot 1`] = `
<div
  className="cocktail-loader"
>
  <h2>
    Loading
  </h2>
</div>
`;

exports[`Cocktail tests snapshots match NavBar matches the snapshot 1`] = `
<nav
  className="cocktail-navbar"
>
  <div
    className="cocktail-nav-center"
  >
    <Link
      to="/Cocktail"
    >
      <img
        alt="cocktail logo"
        className="cocktail-logo"
        src="logo.svg"
      />
    </Link>
    <ul
      className="cocktail-nav-links"
    >
      <li>
        <Link
          to="/Cocktail"
        >
          Home
        </Link>
      </li>
      <li>
        <Link
          to="/cocktail_about"
        >
          About
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Cocktail tests snapshots match SearchForm matches the snapshot 1`] = `
<SearchForm>
  <section
    className="cocktails-section cocktail-search"
  >
    <form
      className="cocktail-search-form"
      onSubmit={[Function]}
    >
      <div
        className="cocktail-form-control"
      >
        <label
          htmlFor="name"
        >
          Search your favorite cocktail
        </label>
        <input
          id="name"
          onChange={[Function]}
          type="text"
        />
      </div>
    </form>
  </section>
</SearchForm>
`;
